src/LibraryManager.Vsix/UI/InstallDialog.xaml.cs

```diff
@@ -175,7 +176,27 @@ private void ProviderComboBox_SelectionChanged(object sender, SelectionChangedEv
 
         private void InstallButton_Clicked(object sender, RoutedEventArgs e)
         {
-            CloseDialog(true);
+            if (!ViewModel.IsLibraryInstallationStateValid())
+            {
+                int result;
+                IVsUIShell shell = VisualStudio.Shell.Package.GetGlobalService(typeof(SVsUIShell)) as IVsUIShell;
+
+                shell.ShowMessageBox(dwCompRole: 0,
+                rclsidComp: Guid.Empty,
+                pszTitle: null,
+                pszText: ViewModel.ErrorMessage,
+                pszHelpFile: null,
+                dwHelpContextID: 0,
+                msgbtn: OLEMSGBUTTON.OLEMSGBUTTON_OK,
+                msgdefbtn: OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST,
+                msgicon: OLEMSGICON.OLEMSGICON_WARNING,
+                fSysAlert: 0,
+                pnResult: out result);
```

Comment:
nit: can you tab in these lines once?

Reply:
Updated in iteration 7

---
In reply to: [201158083](https://github.com/aspnet/LibraryManager/pull/293#discussion_r201158083) [](ancestors = 201158083)